### PR TITLE
Mecha tweaks and fixes

### PR DESCRIPTION
**Pull request by MadmanMartian**
- Removes the redundant manual move-state management that's going on for the Clarke and Honker
 - Hoverpod now just returns 1 on check_for_support, rather than copypasting the dyndomove
 - Slows the clarke way the hell down on turfs, it is still superfast in space
 - Adds a cost to the space movement for clarke
 - Adds iontrails to the clarke

### CODE REVIEW

**Comment by Kurfursten** [comment]
You'll take my Clarke speed when you take MoMMI ventcrawling speed away heathen

**Comment by MadmanMartian** [comment]
Sorry, should've clarified. PR comment updated.

**Comment by Kurfursten** [comment]
90% of engineering that a Clarke is practical for is on-station movement. I use a Clarke to quickly fix a breach and unload air canisters, not to go build an SME line. Waving the switchtool about is very inefficient compared to two handing it.

File: code/game/mecha/working/clarke.dm [hunk 1]
```diff
@@ -14,19 +14,27 @@
...
 /obj/mecha/working/clarke/Destroy()
 	qdel(scrubber)
 	scrubber = null
+	qdel(ion_trail)
+	ion_trail = null
 	..()
 
 /obj/mecha/working/clarke/check_for_support()
-	return 1
+	if(cell.use(20))
```
Comment (DamianX) [inline code/game/mecha/working/clarke.dm]: runtime if cell is null?

**Comment by bathosbathtime** [comment]
Why make the Clarke slow

File: code/game/mecha/working/clarke.dm (hunk 1 above)
Comment (Exxion) [reply code/game/mecha/working/clarke.dm]: Looks like a perfect place for `?.`

Comment (Exxion) [reply code/game/mecha/working/clarke.dm]: Wait I can just do this
```suggestion
	if(cell?.use(20))
```

**Comment by Shadowmech88** [comment]
The purpose of making the Clarke incredibly fast was that, were it not at least very close to the foot speed of a regular human, it would in most circumstances be a downgrade from just running around with a hardsuit and a toolbelt. As a tradeoff for being very fast, it was given 0 offensive capability and armor made of paper.
To be honest, I don't quite remember why I made them so fast in space. Maybe changing their speed didn't occur to me. I admit that having them be so fast and precisely controllable in space making them essentially a smaller spacepod might not be a good idea, so I think it would be a better solution to make them slow in space, and leave their on-turf or turf-adjacent speed the same. If their speed is to be changed at all, that is.
Justify it by saying they use weak ion thrusters to move around in space.

**Pull request closed without merging**